Structure_Engine/Query/HasReinforcement.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.Reinforcement;
+using BH.oM.Structure.SectionProperties;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns true if the Bar has a concrete section with reinforcement intent defined with at least one IBarReinforcement in it. False if not, or if the Bar, ConcreteSection or BarRebarIntent is null.")]
+        [Input("bar", "The Bar to check for reinforcement.")]
+        [Output("hasReinforcement", "Returns true if the Bar has reinforcement assigned, false otherwise.")]
+        public static bool HasReinforcement(this Bar bar)
```

Comment:
```suggestion
        public static bool ExistsReinforcement(this Bar bar)
```

I wonder if this should be as suggested? Do we have a naming convention for this?

Reply:
We have a couple of other `HasXxxxxxx` methods in structure engine, and there are a few more sprinkled around. Also feels like a good couple with the `IsXxxxxx` methods that we have a significant amount of. But you are a native speaker, so happy to be voted down 🛩️ 😄

Reply:
If there's other examples dotted around happy to leave as is.